pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1495,10 +1498,13 @@ static void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     }
     return;
   } else if (PyDelta_Check(obj)) {
-    npy_int64 value =
-        PyObject_HasAttrString(obj, "_value") ? get_long_attr(obj, "_value")
-                                              : // pd.Timedelta object or pd.NaT
-            total_seconds(obj) * 1000000000LL;  // nanoseconds per sec
+    // pd.Timedelta object or pd.NaT should evaluate true here
+    // fallback to nanoseconds per sec for other objects
+    // TODO(anyone): cast below loses precision if total_seconds return
+    // value exceeds number of bits that significand can hold
+    int64_t value = PyObject_HasAttrString(obj, "_value")
+                        ? get_long_attr(obj, "_value")
+                        : (int64_t)total_seconds(obj) * 1000000000LL;
```

Comment:
Does the `int64_t` cast apply to `total_seconds(obj)` first? I think this would truncate e.g. `Timedelta(milliseconds=1)` to 0 first before multiplying by `1000000000LL`?

Reply:
Oh nice catch - I think you are right about that. Let me see if we can add a test for that